scripts/ai/AIDriveStrategyCombineCourse.lua

```diff
@@ -1610,8 +1614,13 @@ end
 --- Register a combine unload AI driver for notification about combine events
 --- Unloaders can renew their registration as often as they want to make sure they remain registered.
 ---@param driver AIDriveStrategyUnloadCombine
+---@return boolean Is valid unloader registered?
 function AIDriveStrategyCombineCourse:registerUnloader(driver)
-    self.unloader:set(driver, 1000)
+    if driver:getUnloadTargetType() == AIDriveStrategyUnloadCombine.UNLOAD_TYPES.COMBINE then

```

Comment:
This is very confusing to me, why would an unloader with silo mode try to register? Doesn't the combine or silo loader call an unloader with the proper unload type anyway?